Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabSaveLoadHandler.cpp

```diff
@@ -129,13 +143,242 @@ namespace AzToolsFramework
+                        // is a procprefab:
+                        AZStd::vector<const ProductAssetBrowserEntry*> products;
+                        entry->GetChildrenRecursively<ProductAssetBrowserEntry>(products);
+                        for (auto product : products)
+                        {
+                            evaluateProduct(product, false);
+                        }
+                    }
+                }
+                if (entry->GetEntryType() == AssetBrowserEntry::AssetEntryType::Product)
+                {
+                    // we'll get here if the user specifically and intentionally selected a product
+                    // in the asset browser instead of a source (ie, selecting an actual model or procprefab
+                    // instead of the fbx that produced it).
+                    evaluateProduct(azrtti_cast<const ProductAssetBrowserEntry*>(entry), true);
+                }
+            }
+
+            return foundSomething;
+        }
+
+         // Drag and drop overrides:
+        int PrefabSaveHandler::GetDragAndDropItemViewEventsPriority() const
         {
-            auto instantiatePrefabOutcome = s_prefabPublicInterface->InstantiatePrefab(sourceFilePath, parentId, position);
+            return 10; // higher than 0 so this class gets ahead of other built in
+        }
+        // listview/outliner dragging:
 
-            if (!instantiatePrefabOutcome.IsSuccess())
+        void PrefabSaveHandler::CanDropItemView(bool& accepted, AzQtComponents::DragAndDropContextBase& context)
+        {
+            if (accepted)
             {
-                WarningDialog("Prefab Instantiation Error", instantiatePrefabOutcome.GetError());
+                return;
+            }
+            using namespace AzToolsFramework;
+            if (EntityOutlinerDragAndDropContext* outlinerContext = azrtti_cast<EntityOutlinerDragAndDropContext*>(&context))
+            {
+                if (CanDragAndDropData(outlinerContext->m_dataBeingDropped))
+                {
+                    accepted = true;
+                }
+            }
+        }
+
+        void PrefabSaveHandler::DoDropItemView(bool& accepted, AzQtComponents::DragAndDropContextBase& context)
+        {
+            using namespace AzToolsFramework;
+            if (accepted)
+            {
+                return;
+            }
+
+            if (EntityOutlinerDragAndDropContext* outlinerContext = azrtti_cast<EntityOutlinerDragAndDropContext*>(&context))
+            {
+                AZStd::vector<AZStd::string> thingsToSpawn;
+
+                AZ::Vector3 spawnLocation = AZ::Vector3::CreateZero();
+                if (!outlinerContext->m_parentEntity.IsValid())
+                {
+                    EditorRequestBus::BroadcastResult(spawnLocation, &EditorRequestBus::Events::GetWorldPositionAtViewportCenter);
+                }
+
+                if (CanDragAndDropData(outlinerContext->m_dataBeingDropped, &thingsToSpawn))
```

Comment:
Any reason why detaching of proc prefabs is not covered here just like dropping onto viewport? Might be good to cover that too to keep behaviors consistent.

Reply:
hmm you're right, it should auto detach using the same rules as the viewport